### ISSUE TITLE
Redis license change

### ISSUE THREAD

**Issue by mieseprem** [nextcloud-snap/nextcloud-snap#2753]
Hello,

are you aware of the change of the license of Redis as of version 7.4? According to [Redis Adopts Dual Source-Available Licensing](https://redis.io/blog/redis-adopts-dual-source-available-licensing/) things will change.
I have no clue how this affects you as open source project or us all as user of this Nextcloud distribution.

I found this license issue described in an Fedora Magazine article: [How to move from Redis to Valkey](https://fedoramagazine.org/how-to-move-from-redis-to-valkey/)

Maybe you spend some minutes and check the impact.

Cheers!

**Comment by scubamuc** [comment]
@mieseprem, thanks for the heads up.

@pachulo will this effect us?

**Comment by kyrofa** [comment]
From their FAQ:

> 17. I am the author of an open-source project that uses Redis in a non-competitive way. If someone else uses my project to produce a competitive product or hosted service (e.g., starts using my project in their SaaS solution), am I at risk of being considered competitive and violating the dual license? Do I need to track all users of my project and report suspected infringing use?
> 
> Only those who are embedding or hosting Redis products in a competitive manner are in violation of the license. The violation does not extend to a project owner who is not doing so and does not require asking others to do so on its behalf. The project owner has no obligation to monitor or report on how others are using their project.

I don't think this will immediately effect us. It's concerning, though. Thankfully we've never really exposed redis in a way that it can be used beyond a Nextcloud cache, so it should be pretty seamless to switch it out with something else.

**Comment by Pilzinsel64** [comment]
> it should be pretty seamless to switch it out with something else

You mean exchange the redis software with something that is able to act like redis? From side of Nextcloud only [this cache methods](https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/caching_configuration.html) are available yet.

Maybe an alternative is using [redis snap](https://snapcraft.io/redis)? Would that be a way around? (sorry, understanding native english license texts are not my best skill 😄)

**Comment by kyrofa** [comment]
> You mean exchange the redis software with something that is able to act like redis?

Right, Valkey is a Redis fork, as an example. However, before we take any action we should probably wait and see what upstream Nextcloud does about this, if anything.

**Comment by Pilzinsel64** [comment]
> see what upstream Nextcloud does about this

I keep my eyes open. ;)

**Comment by scubamuc** [comment]
just as a reminder @pachulo has started a discussion #2663 about using APCu for local cache previously... no need to reinvent the wheel right?

**Comment by Pilzinsel64** [comment]
> Nextcloud upstream recommends ACPu [](https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/caching_configuration.html)

That's true, but as you said, this fits only for `memcache.local`. For `memcache.locking` and `memcache.distributed` it's redis wich is recommended. File locking depends on `memcache.locking` for example.

**Comment by mieseprem** [comment]
Just to have it linked, Upstream has this discussion too: https://github.com/nextcloud/server/issues/44443

**Comment by solracsf** [comment]
Redis 8 is now available under the AGPLv3 open source license...again.

https://redis.io/blog/agplv3/
https://antirez.com/news/151
